snuba/datasets/cdc/groupassignee.py

```diff
@@ -24,11 +25,12 @@ def __init__(
         super().__init__(**kwargs)
         self.__postgres_table = postgres_table
 
-    def get_bulk_loader(self, source: BulkLoadSource, dest_table: str):
+    def get_bulk_loader(self, source: BulkLoadSource, dest_table: str, clickhouse_client: ClickhousePool):
         return SingleTableBulkLoader(
             source=source,
             source_table=self.__postgres_table,
             dest_table=dest_table,
+            clickhouse_client=clickhouse_client,
             row_processor=lambda row: GroupAssigneeRow.from_bulk(row).to_clickhouse(),
```

Comment:
This does not sound right. If the clickhouse pool is now a property of a dataset (indirectly but it is still conceptually a property of the dataset), why would the tableWriter need to be fed with the right clickhouse client. 
At least we should initialize the TableWriter with the right clickhouse_client which is  basically defined there and should not change.
But more broadly I would advise, since, with this change, you are effectively making the clickhouse client a property of the dataset, you try to make the Storage class (which is now committed) provide the reader and the WritableStorage provide the Writer that can be properly initialized inside.
I believe we should have access to the storage and to the dataset in every place where we now need this, but I may be wrong.